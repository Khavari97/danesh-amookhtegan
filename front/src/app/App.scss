@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


