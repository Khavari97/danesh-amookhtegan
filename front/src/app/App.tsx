import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
