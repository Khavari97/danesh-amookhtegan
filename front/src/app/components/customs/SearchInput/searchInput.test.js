import { Grid, Input } from "@material-ui/core";
import React from "react" 
import Enzyme, { shallow,configure,mount } from "enzyme";
import renderer from "react-test-renderer";
import App from "./SearchInput";
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

const setup = (props = {}) => {
  return shallow(<App {...props} />);
};

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  };

describe('render search input component without error',()=>{
    it('it should have value', () => {
        let wrapper = setup();
        wrapper.setProps({
            text: ()=>{},
            default: 'default',
            value: ""})
          expect(wrapper.find(Input).prop('defaultValue')).toBe('default');
      });

    it('test placeholder input',()=>{
        let wrapper = setup();
        let field = findByTestAttr(wrapper, 'placeholder')
      expect(field.text()).toBe('')
    })

    it('render button search',()=>{
        let wrapper = setup();
        let btn = findByTestAttr(wrapper,'btnSearch');
        expect(btn.text()).toBe('جستجو');
        expect(btn.length).toEqual(1);
        expect(btn.prop("color")).toBe("primary");
    })

    it('snapshot',()=>{
        const shot = <App />
        const snapshot = renderer.create(shot).toJSON();
        expect(snapshot).toMatchSnapshot();
    })

})
