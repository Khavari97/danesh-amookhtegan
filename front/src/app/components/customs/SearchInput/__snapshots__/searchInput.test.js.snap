// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render search input component without error snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
  style={
    Object {
      "display": "flex",
      "margin": "5px 0 40px 0",
      "padding": 5,
      "textAlign": "center",
    }
  }
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth"
    data-test="placeholder"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input MuiInput-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="مهارت یا اسم پروژه یا نام کاربری فرد مورد نظر خود را وارد کنید"
      type="text"
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-test="btnSearch"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "marginRight": 5,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      جستجو
      <img
        alt=""
        src="search.svg"
        style={
          Object {
            "marginRight": 5,
          }
        }
        width="100%"
      />
    </span>
  </button>
</div>
`;
